components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -975,7 +975,9 @@ def dump_to_primitive_dict(self):
     def validate_presto_config(cls, values):
         package = values.get("package")
         presto = values.get("presto")
-        query_engine = None
+        query_engine: Optional[QueryEngine] = None
+        # Root validator with `pre=True` is called before other validators, so package has not
+        # been validated as dict yet. Pydantic will throw error later if package is not a dict.
```

Comment:
```suggestion
        # Root validator with `pre=True` is called before other validators, so package has not
        # been validated yet. Pydantic will throw error later if package is invalid.
```
And sorry for one last comment:

what if we do 
```
if not isinstance(package, Package):
    return
query_engine = package.get("query_engine")
```

If package is an instance of Package, it must have query_engine key